enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToMaster.java

```diff
@@ -123,17 +125,26 @@ public URI switchToMaster( LifeSupport haCommunicationLife, URI me )
 
     static URI getMasterUri( URI me, MasterServer masterServer, Config config )
     {
-
         String hostname = config.get( HaSettings.ha_server ).getHost();
-        if ( hostname == null || hostname.contains( "0.0.0.0" ) )
+        InetSocketAddress masterSocketAddress = masterServer.getSocketAddress();
+
+        if ( hostname == null || isWildcard( hostname ) )
         {
-            String masterAddress = ServerUtil.getHostString( masterServer.getSocketAddress() );
-            hostname = masterAddress.contains( "0.0.0.0" ) ? me.getHost() : masterAddress;
+            InetAddress masterAddress = masterSocketAddress.getAddress();
+            hostname = masterAddress.isAnyLocalAddress() ? me.getHost() : ServerUtil.getHostString( masterSocketAddress );
```

Comment:
Does `isAnyLocalAddress` cover the `0.0.0.0` case? The test below suggests that it does, but just want to make sure